### PR TITLE
Feature/apvs 1274/caseworker assignment

### PR DESCRIPTION
**Pull request by msjhall138**
Added assignment to claims and checks to ensure they cannot alter claims without being assigned to them
 
* Added migrations to add AssignedTo and AssignmentExpiry to claims
* Added config for expiry time
* Updated claim index to not show claims assigned to other users - works based off expiry time as well
* Added column to returned tables in searches to say who is assigned to claims
* Added button to assign self on view claim
* Added button so anyone can unassign on view claim
* Added logic to prevent posting when not assigned to claim - throws validation error
* Now hide selects and other posting buttons when not assigned on view claim
* Updated e2e test

## Checklist

- [x] Unit tests
- [x] Code coverage checked
- [x] Coding standards

### CODE REVIEW

File: app/services/data/update-assignment-of-claims.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const config = require('../../../knexfile').intweb
+const knex = require('knex')(config)
+const dateFormatter = require('../date-formatter')
+const environmentVariables = require('../../../config')
+
+module.exports = function (claimId, assignedTo) {
```
Comment (stevenalexander) [inline app/services/data/update-assignment-of-claims.js]: explicitly set assignmentExpiry to null if assigning to null to unassign

File: app/services/data/update-assignment-of-claims.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  return knex('Claim')
+    .where({'ClaimId': claimId})
+    .update({ 'AssignedTo': assignedTo, 'AssignmentExpiry': dateFormatter.now().add(environmentVariables.ASSIGNMENT_EXPIRY_TIME, 'minutes').toDate() })
```
Comment (stevenalexander) [inline app/services/data/update-assignment-of-claims.js]: update LastUpdated so it is checked at the point of assignment to prevent two people assigning themselves and not realising

File: test/integration/services/data/test-get-claims-list-and-count.js [hunk 3]
```diff
@@ -37,7 +38,17 @@ describe('services/data/get-claim-list-and-count', function () {
...
+      return getClaimsListAndCount('TESTING_NONE', false, 0, 10, 'TestUser@test.com')
+        .then(function (result) {
+          expect(result.total.Count).to.equal(0)
+        })
+        .catch(function (error) {
+          throw error
+        })
+    })
+
+    it('should return no data for records assigned to another user', function () {
```
Comment (stevenalexander) [inline test/integration/services/data/test-get-claims-list-and-count.js]: add two new test cases, one for claim AssignedTo is null and one for AssignmentExpiry is past

File: test/integration/services/data/test-update-assignment-of-claims.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      var expiryTime = parseInt(environmentVariables.ASSIGNMENT_EXPIRY_TIME)
+      var currentDate = new Date()
+      var twoMinutesAgo = new Date().setMinutes(currentDate.getMinutes() - (expiryTime - 2))
+      var twoMinutesAhead = new Date().setMinutes(currentDate.getMinutes() + (expiryTime + 2))
+      return updateAssignmentOfClaims(claimId, assignedTo)
+        .then(function () {
+          return knex('Claim').first().where('ClaimId', claimId)
+            .then(function (claim) {
+              expect(claim.AssignedTo).to.equal(assignedTo)
+              expect(claim.AssignmentExpiry).to.be.within(twoMinutesAgo, twoMinutesAhead)
```
Comment (stevenalexander) [inline test/integration/services/data/test-update-assignment-of-claims.js]: add check for LastUpdated
